AppCenterCrashes/AppCenterCrashes/Internals/Util/MSErrorLogFormatter.m

```diff
@@ -529,43 +522,47 @@ + (NSString *)extractExceptionReasonFromReport:(PLCrashReport *)report {
   return exceptionReason;
 }
 
-+ (NSString *)selectorRegisterValueFromReport:(PLCrashReport *)report
-                              ofCrashedThread:(PLCrashReportThreadInfo *)crashedThread
-                                      is64bit:(BOOL)is64bit {
++ (NSString *)selectorRegisterValueFromReport:(PLCrashReport *)report ofCrashedThread:(PLCrashReportThreadInfo *)crashedThread {
 
   /*
    * Try to find the selector in case this was a crash in obj_msgSend.
-   * We search this whether the crash happened in obj_msgSend or not since we
+   * We search this whether the crash happened in objc_msgSend or not since we
    * don't have the symbol!
    */
   NSString *foundSelector = nil;
 
-// Search the registers value for the current architecture.
-#if TARGET_OS_SIMULATOR
-  if (is64bit) {
-    foundSelector = [[self class] selectorForRegisterWithName:@"rsi" ofThread:crashedThread report:report];
-    if (foundSelector == NULL)
-      foundSelector = [[self class] selectorForRegisterWithName:@"rdx" ofThread:crashedThread report:report];
-  } else {
-    foundSelector = [[self class] selectorForRegisterWithName:@"ecx" ofThread:crashedThread report:report];
-  }
-#else
-  if (is64bit) {
-    foundSelector = [[self class] selectorForRegisterWithName:@"x1" ofThread:crashedThread report:report];
-  } else {
+  // Search the registers value for the current architecture.
+  switch (report.machineInfo.processorInfo.type) {
+  case CPU_TYPE_ARM:
     foundSelector = [[self class] selectorForRegisterWithName:@"r1" ofThread:crashedThread report:report];
-    if (foundSelector == NULL)
+    if (foundSelector == NULL) {
       foundSelector = [[self class] selectorForRegisterWithName:@"r2" ofThread:crashedThread report:report];
-  }
-#endif
+    }
+    break;
+
+  case CPU_TYPE_ARM64:
+    foundSelector = [[self class] selectorForRegisterWithName:@"x1" ofThread:crashedThread report:report];
```

Comment:
Just a suggestion: we could simplify the code a bit if we put determination of register name under `switch` and avoid calling the same function 4 times.